pkg/operator/defaults.go

```diff
@@ -75,17 +79,19 @@ var (
 		"v2.54.1",
 		"v2.55.0",
 		"v2.55.1",
-	}
-
-	// Note: Issues in this version won't be supported by operator till its stable
-	// This is only added for users to try the unstable versions.
-	PrometheusExperimentalVersions = []string{
-		"v3.0.0-beta.0",
-		"v3.0.0-beta.1",
-		"v3.0.0-rc.0",
-		"v3.0.0-rc.1",
-		// TODO: To be moved to default latest version once we tested fully.
 		"v3.0.0",
 		"v3.0.1",
 	}
 )
+
+func getLatestPrometheusV2() string {
+	v2 := regexp.MustCompile(`v2`)
+	for i, version := range PrometheusCompatibilityMatrix {
+		match := v2.FindStringIndex(version)
+		// Since first non match would a v3 version
+		if match == nil {
+			return PrometheusCompatibilityMatrix[i-1]
+		}
+	}
+	return ""
```

Comment:
(nit) we can fail hard.
```suggestion
	panic("failed to find a v2.x entry in the compatibility matrix")
```